### ISSUE TITLE
`allow(clippy::all)` does not allow lints as expected

### ISSUE THREAD

**Issue by medwards** [rust-lang/rust-clippy#8104]
### Summary

I am keeping some modules in my tree that I am not maintaining and don't want to touch. I'd like to add `#![allow(clippy::all)]` so I can use clippy and not see those modules, however I still get warnings unless I explicitly disable the lints. I've also tried allowing `rust-2018-idioms` to no effect.

### Reproducer

I tried this code:

```rust
#![allow(clippy::all)]
#![allow(rust_2018_idioms)]
//#![allow(unused_mut)]
//#![allow(dead_code)]

fn gimme_the_vec() -> Vec<usize> {
    let mut foo = Vec::new();
    foo
}
```

I expected to see this happen:
Rust playground output:
```
    Checking playground v0.0.1 (/playground)
    Finished dev [unoptimized + debuginfo] target(s) in 2.25s
```

Instead, this happened:
```
    Checking playground v0.0.1 (/playground)
warning: variable does not need to be mutable
 --> src/lib.rs:7:9
  |
7 |     let mut foo = Vec::new();
  |         ----^^^
  |         |
  |         help: remove this `mut`
  |
  = note: `#[warn(unused_mut)]` on by default

warning: function is never used: `gimme_the_vec`
 --> src/lib.rs:6:4
  |
6 | fn gimme_the_vec() -> Vec<usize> {
  |    ^^^^^^^^^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: `playground` (lib) generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.59s
```

Note that uncommenting the other `allow` statements will silence these warnings.

### Version

```text
Whatever is on `play.rust-lang.org`
```

**Comment by medwards** [comment]
Actually, I now see this is working for `clippy::single_char_pattern`, so its a namespacing thing.

`#![allow(warnings)]` catches all the `rustc` lints.

I'm going to leave the bug open still since I think this is a documentation thing still.

"Allowing/denying lints" should explicitly reference [`rustc` Lint Groups](https://doc.rust-lang.org/rustc/lints/groups.html)

**Comment by llogiq** [comment]
`unused_mut` and `dead_code` are linted by rustc, not clippy. So you'll likely want to `allow` them in those modules in addition to `clippy::all`.
